bwt/bwt.go

```diff
@@ -0,0 +1,227 @@
+package bwt
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+const nullChar = "0"
```

Comment:
Is the null char similar to the "$" I've seen appended to the input string in other implementations?

Reply:
Yes exactly. It was "$" at first. I made it 0 for now because lexicographically, that's the character with the lowest value. This is just so if someone were to try and use the BWT with a string with spaces in it just to mess around and understand how it works, then it would work for them. Some follow up work that should be done:
1. modify the sorting so if we encounter the `nullChar` we always make sure that is the least. Sort everything else lexographically. I can give this a shot in this PR if you want though. I didn't want to add more noise than necessary.
2. Make that nullChar configurable. Neither "$" or "0" should show up in the protein, dna, or rna alphabets but... you never know how users might want to use this.

Reply:
It can be whatever we want now! Replied to your other comment.